app/domain/authentication/constraints/permitted_constraint.rb

```diff
@@ -0,0 +1,23 @@
+module Authentication
+  module Constraints
+
+    # This constraint validates that only the initialized restrictions are permitted.
+    # So, no other restrictions can be found in the given restrictions when calling validate(given_restrictions).
+    class PermittedConstraint
+
+      def initialize(permitted:)
```

Comment:
same here. what do we expect `permitted` to be?

Reply:
Comment added on top of class